### ISSUE TITLE
[FEATURE] Support for re-encrypting/updating a file

### ISSUE THREAD

**Issue by brxken128** [brxken128/dexios#37]
This will be used primarily for updating a file's header version, or just changing the nonce/salt. I do not plan to add support for modifying the AEAD used - users are free to decrypt and re-encrypt to do this, if they so wish.

Use `CryptoParams` arguments (and use the same values within the clap definition of the subcommand)

This will be done via a combined encrypting/decrypting function.

Data will be re-encrypted with the AEAD that the header specifies, which means multiple files should be able to be specified and updated with one command.

The salt and nonces will be changed once re-encrypted.

For streaming, one buffer will be populated with encrypted data, and it will be decrypted to another buffer within memory. It will then be encrypted, and written to the disk. This means no plaintext data will hit the disk, and it will all be kept within memory.

For memory mode, use the same principle.

Ensure that no plain, decrypted data touches the disk (and that all plaintext data is appropriately zeroed out of memory)

I feel as though `dexios update file1.enc file2.enc` is user-friendly enough. Either the user can specify a keyfile/environment key (provided the all of files use that same key), or they should be asked for their key each time (this should also reiterate the file name to prevent any confusion).

**Comment by brxken128** [comment]
I do have a local working implementation of this, however I'm going to save it until after #44 is complete, as this will mean rewriting most of the function anyway.

**Comment by brxken128** [comment]
I'm going to postpone this for now. I don't think this functionality really brings anything to the table, and it's not much easier than running `dexios decrypt` and `dexios encrypt` again. If we were to allow for multiple files to be `updated` at once, we would have to assume that they all use the same key/keyfile.
